hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -207,6 +207,7 @@ public class HoodieWriteConfig extends HoodieConfig {
   public static final ConfigProperty<String> BASE_PATH = ConfigProperty
       .key("hoodie.base.path")
       .noDefaultValue()
+      .makeEssential(true)
```

Comment:
IMO, by defualt, the `essential` property should be false, meaning that the config has to be marked "essential" explicitly, enforcing us to think twice when introducing a new essential config.  For this API, we can remove the argument, i.e., `.makeEssential()` instead of `.makeEssential(true)`.

Reply:
As discussed, we'll use `advanced` and `.markAdvanced()` to be aligned with database advanced configs.